### ISSUE TITLE
Possible fix for TP-Link TL-WR1043N/ND v1 instability

### ISSUE THREAD

**Issue by kriegste** [freifunk-berlin/firmware#555]
WiFi on my TP-Link TL-WR1043N/ND v1 gets inaccessible and needs rebooting after high load situations.

Executing
`echo 0 > /sys/kernel/debug/ieee80211/phy0/ath9k/ani`
upon start seems to solve the issue. I am still monitoring, but after one week of testing, everything seems to be fine.

**Comment by booo** [comment]
Can you provide us with a bit of background information about the option and why it should solve the problem?

**Comment by kriegste** [comment]
ANI is "Adaptive Noise Immunity", but I have no clue if the bug is in the chip or the driver. Fact is, the problem is gone. My router is in a usable state finally.

It was mentioned here:
https://github.com/libremesh/lime-packages/issues/144
https://www.reddit.com/r/linux/comments/203sn0/this_bug_has_been_around_for_years/
https://www.gargoyle-router.com/phpbb/viewtopic.php?t=4667
https://www.walterschlag.net/?paged=16
https://tweakers.net/downloads/42463/lede-17013.html

**Comment by bobster-galore** [comment]
I can confirm that the TP-Link TL-WR1043N/ND v1 will collapse under high wifi load, this is why I replaced it by another router.
Should be 
> echo 0 > /sys/kernel/debug/ieee80211/phy0/ath9k/ani

the standard for this router?

**Comment by torte71** [comment]
As a sidenote: My 1043v2 was not affected by that ani change, neither positive nor negative, wifi still stopped working afters hours under high load / many users.
But it is a fix I read several times about, and it seems to help under certain conditions (maybe only on that router version? I can't remember reading about that problem/fix being version specific).
So I'd vote for including that change for 1043 models.

**Comment by kriegste** [comment]
Uptime now 19 days, still no problems (v1). Before that fixed I needed to reboot up to twice a day, depending on the load.

**Comment by vzsze** [comment]
Turning off ANI seems to have decreased load and stability on one of my TL-WR1043N/ND v1. The other seems to be unaffected, but already showed low load without it. See attached graph.

![graph php](https://user-images.githubusercontent.com/541560/41399711-55ee0d80-6fbb-11e8-8aeb-a24ebc6b44c2.png)

**Comment by EG-freifunk** [comment]
I tried on my 1043v1
`cat /sys/kernel/debug/ieee80211/phy0/ath9k/ani`
Got:
```
            ANI: ENABLED
      ANI RESET: 1182
     OFDM LEVEL: 5
      CCK LEVEL: 2
        SPUR UP: 3201
      SPUR DOWN: 3201
 OFDM WS-DET ON: 25
OFDM WS-DET OFF: 25
     MRC-CCK ON: 0
    MRC-CCK OFF: 0
    FIR-STEP UP: 3235
  FIR-STEP DOWN: 2967
 INV LISTENTIME: 1
    OFDM ERRORS: 76183809
     CCK ERRORS: 1521071
```
There is a plenty of OFDM errors. What the flame this means?

**Comment by vzsze** [comment]
This is the unaffected node, showing low load before and after turning off ANI.
![graph kakao](https://user-images.githubusercontent.com/541560/41406114-d47f0b12-6fcb-11e8-9e7a-805bf028fccc.png)

**Comment by booo** [comment]
Is this something we should discuss with the upstream project (openwrt)? Is there already a discussion upstream? If not can someone open a ticket?

**Comment by FreifunkUFO** [comment]
of course there are many ticket about that problem on other projects using openwrt (libremesh, freifunk-gluon, freifunk-meshkit, battlemesh) ... and also on openwrt (subject may differ, because atm there is no tested or acceptable fix), f.i.

https://bugs.openwrt.org/index.php?do=details&task_id=863

see also: ANI, ani-noise-immunity, weak signal detection, ofdmNoiseImmunityLevel, ATH9K_ANI_OFDM_DEF_LEVEL

**Comment by matthias-klose** [comment]
> I can confirm that the TP-Link TL-WR1043N/ND v1 will collapse under high wifi load, this is why I replaced it by another router.
> Should be
> 
> > echo 0 > /sys/kernel/debug/ieee80211/phy0/ath9k/ani
> 
> the standard for this router?

yes. i am running two 1043NDv1 and they do perform better and more stable with deactivated ANI run by rc.local at boot.
